module/zfs/vdev.c

```diff
@@ -6095,6 +6103,15 @@ vdev_prop_set(vdev_t *vd, nvlist_t *innvl, nvlist_t *outnvl)
 			}
 			vd->vdev_slow_io_t = intval;
 			break;
+		case VDEV_PROP_QUEUE_IO:
+			if (nvpair_value_uint64(elem, &intval) != 0) {
+				error = EINVAL;
+				break;
+			}
+			if (vd->vdev_ops->vdev_op_leaf) {
```

Comment:
Only allowing this to be set on leaves has the complication that when the drive is replaced it will revert back to the default value for the property.  The `VDEV_PROP_CHECKSUM/IO/SLOW_N/T` properties handle this with a workaround which inherits non-default values if they're set on any of the parents (see `vdev_prop_get_inherited`).  We should consider doing the same thing.  The only wrinkle would be you'd want to cache to value in the vdev_t.

Reply:
It looks like `vdev_prop_get_inherited` still needs to be updated to handle inheritance here.